File: 58. Length of Last Word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        s = s.rstrip()
        if not s:
            return 0
        for i in xrange(len(s)-1, -1, -1):
            if s[i] == ' ':
                return len(s) - i - 1
            if i == 0:
                return len(s) - i

File: 58. Length of Last Word.cpp
static inline void rtrim(string &s) {
    s.erase(find_if(s.rbegin(), s.rend(),
        not1(ptr_fun<int, int>(isspace))).base(), s.end());
}

class Solution {
public:
    int lengthOfLastWord(string s) {
        rtrim(s);
        for (int i = s.size()-1; i >= 0; i--)
        {
            if (s[i] == ' ')
                return s.size() - i - 1;
            if (i == 0)
                return s.size() - i;
        }
        return 0;
    }
};
